test/models/question_test.rb

```diff
@@ -14,34 +15,41 @@ class QuestionTest < ActiveSupport::TestCase
   end
 
   test 'latest versions' do
-    assert_equal 4, Question.latest_versions.count
+    assert Question.latest_versions.count < Question.all.count
+  end
+
+  test 'versions must be unique' do
+    q1 = Question.new(content: 'content')
+    assert q1.save
+    q2 = Question.new(content: 'content', version: 1, version_independent_id: q1.version_independent_id)
+    assert_not q2.save
+    q2.version = 2
+    assert q2.save
   end
 
   test 'search' do
-    assert 4, Question.count
     found = Question.search('gender')
     assert 1, found.count
```

Comment:
If someone adds another question with the word 'gender' in it to the fixtures, this is going to come back with a value other than 1, breaking this test.

Also, unless the found variable is used somewhere else, isn't it superfluous?